src/GovUk.Education.ExploreEducationStatistics.Content.Model.Tests/Extensions/ReleaseAmendmentExtensionsTests.cs

```diff
@@ -0,0 +1,1155 @@
+                },
+            };
+
+            var section1Id = Guid.NewGuid();
+
+            release.Content = new List<ReleaseContentSection>
+            {
+                new()
+                {
+                    Release = release,
+                    ReleaseId = release.Id,
+                    ContentSectionId = section1Id,
+                    ContentSection = new ContentSection
+                    {
+                        Id = section1Id,
+                        Heading = "Section 1",
+                        Content = new List<ContentBlock>
+                        {
+                            contentBlock1,
+                            contentBlock2,
+                            contentBlock3,
+                            contentBlock4
+                        },
+                    }
+                },
+            };
+
+            var createdDate = DateTime.Now;
+            var createdById = Guid.NewGuid();
+
+            var amendment = release.CreateAmendment(createdDate, createdById);
+
+            var amendmentDataBlock1 = Assert.IsType<DataBlock>(amendment.ContentBlocks[0].ContentBlock);
+            var amendmentDataBlock2 = Assert.IsType<DataBlock>(amendment.ContentBlocks[1].ContentBlock);
+
+            Assert.NotEqual(dataBlock1.Id, amendmentDataBlock1.Id);
+            Assert.NotEqual(dataBlock2.Id, amendmentDataBlock2.Id);
+
+            var section1 = amendment.Content[0].ContentSection;
+
+            var amendmentContentBlock1 = Assert.IsType<HtmlBlock>(section1.Content[0]);
+            var amendmentContentBlock2 = Assert.IsType<HtmlBlock>(section1.Content[1]);
+            var amendmentContentBlock3 = Assert.IsType<HtmlBlock>(section1.Content[2]);
+            var amendmentContentBlock4 = Assert.IsType<HtmlBlock>(section1.Content[3]);
+
+            Assert.Equal(
+                $"Content block 1 http://localhost/fast-track/{amendmentDataBlock1.Id}",
+                amendmentContentBlock1.Body
+            );
+            Assert.Equal(
+                $"Content block 2 http://localhost/fast-track/{amendmentDataBlock2.Id}/ some other text",
+                amendmentContentBlock2.Body
+            );
+
+            Assert.Equal(
+                $"<p>Content block 3 <a href=\"http://localhost/fast-track/{amendmentDataBlock1.Id}\">link text</a></p>",
+                amendmentContentBlock3.Body
+            );
+            Assert.Equal(
+                $@"
+                    <p>Content block 4 http://localhost/fast-track/{amendmentDataBlock1.Id} http://localhost/fast-track/{amendmentDataBlock2.Id}</p>
+                    <p><a href=""http://localhost/fast-track/{amendmentDataBlock1.Id}"">link 1 text</a></p>
+                    <p><a href=""http://localhost/fast-track/{amendmentDataBlock2.Id}/"">link 2 text</a></p>
+                    ",
+                amendmentContentBlock4.Body
+            );
+        }
+
+        [Fact]
+        public void CreateAmendment_StripsComments()
```

Comment:
I see we end up doing similar testing here between creating release amendments and publishing release content. I don't mean in terms of the content filtering functionality being called, but in terms of checking that comments in different formats are being removed correctly. I suppose the solution would be to abstract out something responsible for stripping comments that can be tested on its own and then just check it's used in both places.

Reply:
Good point! I've refactored the comments regex into some utils and now just test that more extensively in separate tests.

The tests for the publishing content/amendment now just do some basic checks around this too to make sure it mostly works.